crypto/provider_core.c

```diff
@@ -608,8 +617,13 @@ const OSSL_ALGORITHM *ossl_provider_query_operation(const OSSL_PROVIDER *prov,
  * never knows.
  */
 static const OSSL_ITEM param_types[] = {
-    { OSSL_PARAM_UTF8_PTR, "openssl-version" },
-    { OSSL_PARAM_UTF8_PTR, "provider-name" },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_OPENSSL_VERSION },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_PROV_NAME },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_MODULE_FILENAME },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_MODULE_MAC },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_INSTALL_VERSION },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_INSTALL_MAC },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_INSTALL_STATUS },
```

Comment:
I'm unsure about the added names here, for the simple reason that they aren't anything that the provider functionality offers *internally*, but are rather stuff it receives from external sources.

Quite frankly, the get_types function needs a review, perhaps even its format.

Reply:
Not even sure its really needed.